File: BiteClub/Models/FriendRequest.swift
import Foundation

enum RequestStatus: String, Codable {
    case pending, accepted, rejected
}

struct FriendRequest: Codable, Hashable, Identifiable {
    var id: UUID
    var senderId: UUID
    var receiverId: UUID
    var status: RequestStatus
    var createdAt: Date
    var updatedAt: Date
    
    var sender: User?
    var receiver: User?
    
    enum CodingKeys: String, CodingKey {
        case id = "id"
        case senderId = "sender_id"
        case receiverId = "receiver_id"
        case status = "status"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case sender = "sender"
        case receiver = "receiver"
    }
}

File: BiteRunr/shared/src/commonMain/kotlin/org/biterunr_studios/biterunr/Models/FriendRequest.kt
package org.biterunr_studios.biterunr.Models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.datetime.Instant

@Serializable
enum class RequestStatus {
    @SerialName("pending")
    PENDING,

    @SerialName("accepted")
    ACCEPTED,

    @SerialName("rejected")
    REJECTED
}


@Serializable
data class FriendRequest(
    @SerialName("id")
    val id: String,

    @SerialName("sender_id")
    val senderId: String,

    @SerialName("receiver_id")
    val receiverId: String,

    @SerialName("status")
    val status: RequestStatus,

    @SerialName("created_at")
    val createdAt: Instant,

    @SerialName("updated_at")
    val updatedAt: Instant,

    @SerialName("sender")
    val sender: UserProfile? = null,

    @SerialName("receiver")
    val receiver: UserProfile? = null
)
